code/controllers/subsystem/economy.dm

```diff
@@ -0,0 +1,316 @@
+		"stagnant_cash" = list(),     //How many credits are sitting in stagnant accounts this round
+		"non_stagnant_cash" = list()  //How many credits are in active accounts this round
+	)
+	///time to next stats check
+	var/next_data_check = 0
+
+
+	//////CARGO VARIABLES/////
+	///the department account tethered to this supply console, we keep a ref here for shuttle operations
+	var/datum/money_account/cargo_account
+	///Current Order number
+	var/ordernum = 1
+
+	/// points gained per slip returned
+	var/credits_per_manifest = 5
+	/// points gained per crate returned
+	var/credits_per_crate = 15
+	/// points gained per intel returned
+	var/credits_per_intel = 750
+	/// points gained per plasma returned
+	var/credits_per_plasma = 10
+	/// points gained per research design returned
+	var/credits_per_design = 20
+
+	/// Remarks from Centcom on how well you checked the last order.
+	var/centcom_message
+	/// Typepaths for unusual plants we've already sent CentComm, associated with their potencies
+	var/list/discovered_plants = list()
+	var/list/tech_levels = list()
+	var/list/research_designs = list()
+
+	///Requested crates, waiting for approval by department heads
+	var/list/request_list = list()
+	///Approved Crates, waiting to be delivered
+	var/list/shopping_list = list()
+	///Crates that will be on next shuttle
+	var/list/delivery_list = list()
+
+	///Full list of all available supply packs to purchase
+	var/list/supply_packs = list()
+	var/sold_atoms = ""
+
+	var/list/all_supply_groups = list(
+		SUPPLY_EMERGENCY,
+		SUPPLY_SECURITY,
+		SUPPLY_ENGINEER,
+		SUPPLY_MEDICAL,
+		SUPPLY_SCIENCE,
+		SUPPLY_ORGANIC,
+		SUPPLY_MATERIALS,
+		SUPPLY_MISC,
+		SUPPLY_VEND
+	)
+
+	//////Paycheck Variables/////
+	/// time to next payday
+	var/next_paycheck_delay = 0
+	/// total paydays this round
+	var/payday_count = 0
+
+	var/global_paycheck_bonus = 0
+	var/global_paycheck_deducation = 0
+
+/datum/controller/subsystem/economy/vv_edit_var(var_name, var_value)
+	switch(var_name)
+		//These are all things that admins should not be touching during production, these are either used for logging
+		//or economy critical things that should not be touched
+		if("payday_count")
+			return FALSE //fuck off, used for logging
+		if("sold_atoms")
+			return FALSE //fuck off, used for logging
+		if("cargo_account")
+			if(!istype(var_value, /datum/money_account))
+				return FALSE //really fuck off, you're vv editing something to a value that will break the economy
+	return ..()
+
+/datum/controller/subsystem/economy/Initialize()
+	///create main station accounts
+	if(!GLOB.current_date_string)
+		GLOB.current_date_string = "[time2text(world.timeofday, "DD Month")], [GLOB.game_year]"
+	if(GLOB.station_money_database)
+		populate_station_database()
+		cargo_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_SUPPLY)
+		if(!cargo_account)
+			WARNING("SSeconomy could not locate the supply department account")
```

Comment:
shouldnt this warning be 1 indent lower?

Reply:
fixed in next commit